### PR TITLE
Move updaters into definitions subdir

### PR DESCRIPTION
**Pull request by colinking**
## Description

The Views definition updater will use the same setup as the JavaScript task updater, which currently lives in the (task-specific) runtime. I extracted the updaters into `pkg/definitions/updaters` which is not task-specific. I'll add Views here in a follow-up PR.

There are no functional code changes beyond moving and renaming.

## Test plan

Tests should pass.

### CODE REVIEW

**Comment by colinking** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1107** <a href="https://app.graphite.dev/github/pr/airplanedev/cli/1107" target="_blank">Graphite</a> 
    * **PR #1108** <a href="https://app.graphite.dev/github/pr/airplanedev/cli/1108" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/airplanedev/cli/1108?utm_source=stack-comment).

File: .github/workflows/javascript.yml [hunk 1]
```diff
@@ -13,30 +13,30 @@ concurrency:
...
   run:
-    name: test (${{ matrix.dir }})
+    name: test (${{ matrix.dir.name }})
     strategy:
       fail-fast: false
       matrix:
         dir:
-          - pkg/runtime/javascript/updater
-          - pkg/deploy/discover/parser/node
+          - { name: "updater", value: "pkg/definitions/updaters/javascript" }
```
Comment (colinking) [inline .github/workflows/javascript.yml:21]: I'll update the required CI checks before merging.

File: pkg/definitions/updaters/javascript.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	"github.com/pkg/errors"
+)
+
+//go:embed javascript/index.js
+var javascriptUpdater []byte
+
+var airplaneErrorRegex = regexp.MustCompile("__airplane_error (.*)\n")
+var airplaneOutputRegex = regexp.MustCompile("__airplane_output (.*)\n")
+
+func UpdateJavaScriptTask(ctx context.Context, logger logger.Logger, path string, slug string, def definitions.Definition) error {
```
Comment (francescaleung) [inline pkg/definitions/updaters/javascript.go:24]: why not UpdateNodeTask? we don't do something different for typescript, do we? I guess I'm mostly confused on when we want to use JavaScript vs Node in our internal naming.

**Review by francescaleung — APPROVED** [review]
looks like you need to fix your tests

**Pull request merged**